File: BE/vnpay/src/main/java/com/vnpay/entity/Product.java
package com.vnpay.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.math.BigDecimal;
import java.time.Instant;

@NoArgsConstructor
@SuperBuilder
@Data
@Entity
@Table
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, length = 255)
    private String name;

    @Column(length = 255)
    private String description;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal price;

    @Column(nullable = false)
    private Integer discount;

    @Column(length = 255)
    private String image;

    @Column(nullable = false)
    private Boolean available;

    @Column(name = "created_date", nullable = false)
    private Instant createdDate;
}

File: FE/vnpay/src/model/Product.ts
import { Status } from "../enum/status";

/**
 * Represents a product's information in the response.
 */
export interface ProductResponse {
    id: number;
    name: string;
    description: string;
    price: number;
    discount: number;
    image: string;
    available: boolean;
    createdDate: string;  
}

/**
 * Represents an order's information in the response.
 */
export interface OrderDtoResponse {
    id: number;
    orderId: string;
    totalAmount: number; 
    status: Status;
    createdDate: string; 
}